packages/node/src/behavior/state/validation/ValueValidator.ts

```diff
@@ -221,11 +221,22 @@ function createSimpleValidator(
 }
 
 function createIntegerValidator(schema: ValueModel, supervisor: RootSupervisor, name: string) {
+    const nullable = schema.effectiveQuality.nullable;
     let assertion;
-    if (schema.effectiveQuality.nullable) {
-        assertion = assertInt.nullable[name];
-    } else {
-        assertion = assertInt.notNullable[name];
+    // Let's check schema type specific assertations first
+    if (schema.type === "epoch-s" || schema.type === "epoch-us") {
```

Comment:
Ideally these would probably be "date" metatype instead of integer which would change to entirely different visitation.  Probably too involved though because date handling would then be more modifications...  Maybe could just leave a TODO